engine/src/exec-dialog.cpp

```diff
@@ -711,10 +711,25 @@ void MCDialogExecCustomAskDialog(MCExecContext& ctxt, MCNameRef p_stack, MCNameR
 	t_success = MCStringCreateMutable(0, &t_arg_string) &&
 	MCStringAppendFormat(*t_arg_string, "ask %@", p_type);
 	
+    /* Custom ask dialog parameters are '\0' separated. Prior to 7, any arguments
+     * containing '\0' would be truncated at the NUL. However, since 7 such strings
+     * have caused extra arguments to be passed. Therefore, we revert to the pre-7
+     * behavior and truncate. */
 	for (uindex_t i = 0; t_success && i < p_arg_count; i++)
-		t_success = MCStringAppendNativeChar(*t_arg_string, '\0') &&
-		(p_args[i] == nil || MCStringAppend(*t_arg_string, p_args[i]));
-	
+    {
+        t_success = MCStringAppendNativeChar(*t_arg_string, '\0');
+        if (t_success && p_args[i] != nullptr)
+        {
+            MCRange t_range;
+            if (!MCStringFirstIndexOfChar(p_args[i], '\0', 0, kMCStringOptionCompareExact, t_range.length))
+            {
+                t_range.offset = 0;
+                t_range.length = UINDEX_MAX;
+            }
+            t_success = MCStringAppendSubstring(*t_arg_string, p_args[i], t_range);
```

Comment:
This looks like t_range.offset will be uninitialised if MCStringFirstIndexOfChar succeeds